tests/post/steps/test_authentication.py

```diff
@@ -254,4 +287,16 @@ def _dex_auth_request(control_plane_ip, username, password):
     return result
 
 
+def _get_pod_name(k8s_client, ssh_config, label, namespace):
+    try:
+        pods = kube_utils.get_pods(
+            k8s_client, ssh_config, label, namespace=namespace
+        )
+    except Exception as exc:
+        pytest.fail("Unable to get pod name with error: {}".format(exc))
+
+    for pod in pods:
+        return pod.metadata.name
```

Comment:
Not fond of this. It's misleading, as you're getting "the name of the first Pod that matched this query", not "the Pod name". Why not check all Pods?